scripts/receipts.sh

```diff
@@ -77,31 +88,89 @@ stack.
   --run-image     -r  path to OCI image of build image
-                      ${BUILD_DIR}/run.oci
+                      ${DEFAULT_BUILD_DIR}/run.oci
   --build-receipt -B  path to output build image package receipt. Defaults to
-                      ${BUILD_DIR}/build-receipt.txt
+                      ${DEFAULT_BUILD_DIR}/build-receipt.cyclonedx.json
   --run-receipt   -R  path to output run image package receipt. Defaults to
-                      ${BUILD_DIR}/run-receipt.txt
+                      ${DEFAULT_BUILD_DIR}/run-receipt.cyclonedx.json
 USAGE
 }
 
 function tools::install() {
+  util::tools::crane::install \
+    --directory "${BIN_DIR}"
+  util::tools::jam::install \
+    --directory "${BIN_DIR}"
   util::tools::syft::install \
     --directory "${BIN_DIR}"
 }
 
-function receipts::generate() {
-  local image output
-
-  image="${1}"
-  output="${2}"
-
-  util::print::title "Generating package SBOM for ${image}"
+# Generates syft receipts for each architecture for given oci archives
+function receipts::generate::multi::arch() {
+  local buildArchive runArchive
+  local registryPort registryPid localRegistry
+  local imageType archiveName imageReceipt receiptFilenames
+
+  receiptFilenames=""
+  buildArchive="${1}"
+  runArchive="${2}"
+  buildOutput="${3}"
+  runOutput="${4}"
+  build_dir="${5}"
+
+  registryPort=$(get::random::port)
+  registryPid=$(local::registry::start $registryPort)
+  localRegistry="127.0.0.1:$registryPort"
+
+  # Push the oci archives to the local registry
+  jam publish-stack \
+    --build-ref "$localRegistry/build" \
+    --build-archive $buildArchive \
+    --run-ref "$localRegistry/run" \
+    --run-archive $runArchive >/dev/null
+
+  # Ensure we can write to the build_dir
+  if [ $(stat -c %u $build_dir) = "0" ]; then
+    sudo chown -R "$(id -u):$(id -g)" "$build_dir"
+  fi
+
+  for archivePath in "${buildArchive}" "${runArchive}" ; do
+    archiveName=$(basename "${archivePath}")        # either 'build.oci' or 'run.oci'
+    imageType=$(basename -s .oci "${archivePath}")  # either 'build' or 'run'
+
+    util::print::title "Generating package SBOM for ${archiveName}"
+
+    for imageArch in $(crane manifest "$localRegistry/$imageType" | jq -r '.manifests[].platform.architecture'); do
+
+      if [[ "$imageType" = "build" ]]; then
+        dir=$(dirname ${buildOutput})
+        fileName=$(basename ${buildOutput})
+      elif [[ "$imageType" = "run" ]]; then
+        dir=$(dirname ${runOutput})
+        fileName=$(basename ${runOutput})
+      fi
+
+      if [ $imageArch = "amd64" ]; then
+        imageReceipt="${dir}/${fileName}"
+      else
+        imageReceipt="${dir}/${imageArch}-${fileName}"
+      fi
+
+      util::print::info "Generating CycloneDX package SBOM using syft for $archiveName on platform linux/$imageArch saved as $imageReceipt"
+
+      # Generate the architecture-specific SBOM from image in the local registry
+      syft scan "registry:$localRegistry/$imageType" \
```

Comment:
This came from the jammy-tiny-stack implementation, but instead of using `package` argument like so
https://github.com/paketo-buildpacks/jammy-tiny-stack/blob/50305798894de776ac83d469ef693f1d99a62b49/scripts/receipts.sh#L145-L149

i replaced it with the `scan` argument, as the `package` argument is deprecated and it throws a warning.